src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java

```diff
@@ -275,6 +439,86 @@ public IBlockState getStateForPlacement(
     @Override
     protected BlockStateContainer createBlockState()
     {
-        return new BlockStateContainer(this, FACING);
+        return new BlockStateContainer(this, new IProperty[] {FACING, VARIANT});
+    }
+
+    /**
+     * Types that the {@link BlockConstructionTape} supports
+     */
+    public enum Type implements IStringSerializable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 59 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)